src/operations/explainable_command.ts

```diff
@@ -0,0 +1,44 @@
+import { CommandOperation, OperationParent, CommandOperationOptions } from './command';
+import { Explain, Verbosity, VerbosityLike } from '../explain';
+import { Callback, Document, MongoError, Server } from '..';
+
+/** @public */
+export interface ExplainOptions {
+  explain?: VerbosityLike;
+}
+
+/** @internal */
+export abstract class ExplainableCommand<
+  T extends ExplainOptions & CommandOperationOptions,
+  TResult = Document
+> extends CommandOperation<T, TResult> {
+  explain?: Explain;
```

Comment:
It's my impression that all commands are, or will be explainable, shouldn't we put the explain logic directly into CommandOperation then? (Maybe that assumption is wrong)

Reply:
Not sure about the long term plan of whether all commands will be explainable. At least for now, explain is not supported on `insert`, for example, so `insertOne` cannot be explained. This makes sense to me; since I'm not sure what explain results could even look like for an insert. 

That said, we could roll this logic into the `CommandOperation` class and just not expose explain options on the operations that are not explainable.

Reply:
Aspects would be a good fit for explainable support we can remove the inheritance here and rely on the aspects presence to permit or remove explainability from command construction 

(chatted offline)